misc/discv5/src/query_pool/peers/closest.rs

```diff
@@ -62,65 +64,34 @@ pub struct QueryConfig {
     /// for before it terminates. Defaults to the maximum number of entries in a
     /// single k-bucket, i.e. the `k` parameter in the Kademlia paper.
     pub num_results: usize,
+
+    /// The timeout for a single peer.
+    ///
+    /// If a successful result is not reported for a peer within this timeout
+    /// window, the iterator considers the peer unresponsive and will not wait for
+    /// the peer when evaluating the termination conditions, until and unless a
+    /// result is delivered. Defaults to `10` seconds.
+    pub peer_timeout: Duration,
 }
 
-impl Default for QueryConfig {
+impl Default for FindNodeQueryConfig {
```

Comment:
I think potentially, we remove the `Default` impl and instead create a `new_from_config(config: &Discv5Config)` and we already set the defaults in the config and should make the code a bit nicer.

This way we won't have mismatched default values